scripts/ref-resolver.js

```diff
@@ -1,112 +1,129 @@
-var parser = require("@apidevtools/json-schema-ref-parser")
+var parser = require("@apidevtools/json-schema-ref-parser");
```

Comment:
why are writing this logic ourselves , shouldn't the openapi / jsonschema toolchain automatically resolve the refs

Reply:
It was already in place, i just added the x-order logic..must be the formatter showing whole change. 

Answering the question, I tried directly running the codegen it fails to resolve certain refs and has something to do with nested level.

Reply:
here is the error:
```bash
error generating code: error generating type definitions: error generating Go types for component schemas: error converting Schema model_definition to Go type: error generating Go schema for property 'category': error generating Go schema for property 'id': error turning reference (../core.json#/definitions/uuid) into a Go type: unexpected reference depth: 3 for ref: #/definitions/uuid local: false
```